debian/control

```diff
@@ -107,6 +107,8 @@ Build-Depends: automake,
                xmlstarlet <pkg.ceph.check>,
                nasm [amd64],
                zlib1g-dev,
+	       libarrow-dev,
+	       libparquet-dev,
```

Comment:
So the problem you're running into in the Jenkins job,

```
mk-build-deps: Unable to install ceph-build-deps at /usr/bin/mk-build-deps line 457.
mk-build-deps: Unable to install all build-dep packages
Build step 'Execute shell' marked build as failure
```

Is these packages `libarrow-dev` and `libparquet-dev` aren't available in the standard Ubuntu repos.

I really prefer not to add another third-party repo to `install-deps.sh` if we can avoid it.

Will these build dependencies be required for every single Ceph build or could we make them optional like we do `with_seastar` and `with_jaeger`?

Reply:
> So the problem you're running into in the Jenkins job,
> 
> ```
> mk-build-deps: Unable to install ceph-build-deps at /usr/bin/mk-build-deps line 457.
> mk-build-deps: Unable to install all build-dep packages
> Build step 'Execute shell' marked build as failure
> ```
> 
> Is these packages `libarrow-dev` and `libparquet-dev` aren't available in the standard Ubuntu repos.
> 
> I really prefer not to add another third-party repo to `install-deps.sh` if we can avoid it.
> 
> Will these build dependencies be required for every single Ceph build or could we make them optional like we do `with_seastar` and `with_jaeger`?

https://arrow.apache.org/install/ contains packages also for ubuntu

yes, we can avoid that(adding third-party repo)

It should be optional, the user should decide to install the arrow library.
**question**: upon running tuethology, will it install arrow libraries?
otherwise, it will cause failure.

Reply:
Are the libraries *build* time or *run* time dependencies?  Or both?

Reply:
both

Reply:
Okay.  It looks like I already added the Arrow repo to the testnodes so we just need to figure out getting the Arrow repo on the build hosts.  I'd like it to only be enabled when required instead of all the time.

What I'd propose is add a `<pkg.ceph.arrow>` tag next to the arrow deps in `debian/control`.  See https://github.com/ceph/ceph/blob/master/debian/control#L32 for an example.

Then, in install-deps.sh, [here](https://github.com/ceph/ceph/blob/master/install-deps.sh#L357),
```
	if $with_arrow; then
	    build_profiles+=",pkg.ceph.arrow"
            wget -qO - https://dist.apache.org/repos/dist/dev/arrow/KEYS | sudo apt-key add -
            echo "deb [arch=amd64] http://dl.bintray.com/apache/arrow/ubuntu $codename main" | sudo tee /etc/apt/sources.list.d/arrow.list
	fi
```

Then later after https://github.com/ceph/ceph/blob/master/install-deps.sh#L363,
```
rm -f /etc/apt/sources.list.d/arrow.list
```

And finally, you'll need to find a way to set `with_arrow=true`.  Do you have an idea on how we can set that a build time so it only gets set to `true` if the arrow deps are actually needed?

Reply:
> Okay. It looks like I already added the Arrow repo to the testnodes so we just need to figure out getting the Arrow repo on the build hosts. I'd like it to only be enabled when required instead of all the time.
> 
> What I'd propose is add a `<pkg.ceph.arrow>` tag next to the arrow deps in `debian/control`. See https://github.com/ceph/ceph/blob/master/debian/control#L32 for an example.
> 
> Then, in install-deps.sh, [here](https://github.com/ceph/ceph/blob/master/install-deps.sh#L357),
> 
> ```
> 	if $with_arrow; then
> 	    build_profiles+=",pkg.ceph.arrow"
>             wget -qO - https://dist.apache.org/repos/dist/dev/arrow/KEYS | sudo apt-key add -
>             echo "deb [arch=amd64] http://dl.bintray.com/apache/arrow/ubuntu $codename main" | sudo tee /etc/apt/sources.list.d/arrow.list
> 	fi
> ```
> 
> Then later after https://github.com/ceph/ceph/blob/master/install-deps.sh#L363,
> 
> ```
> rm -f /etc/apt/sources.list.d/arrow.list
> ```
> 
> And finally, you'll need to find a way to set `with_arrow=true`. Do you have an idea on how we can set that a build time so it only gets set to `true` if the arrow deps are actually needed?

it a question also for @cbodley, @mattbenjamin, more features depending on the arrow library are in plan. 
the majority of the users do not need to build with arrow libraries.


since the teuthology contain tests(runtime) for arrow functionalities, the shaman should always build with arrow(as currently done).

as for the build host, no need to build arrow all the time.
is it possible to set `with_arrow` to `true`, upon some file exist/or contain some value? 
this file will be deleted upon build is done(PR is done)?

Reply:
What do you mean, no need to have arrow built all the time?  It would indeed be good if we could actually *build* it only when we need to update it.  However, I think rgw will normally need it built.

Reply:
> I think rgw will normally need it built.

Okay.  That's what I was getting at.

I'll just add the arrow repo to all the build hosts.

Reply:
make sure.
most users (fedora,ubuntu,centos) do not need arrow.

do they need to install arrow on their **private machines**?.(using install-deps.sh)
will it apply to all distributions?

Reply:
Good point.  Yes, any machine that builds Ceph after this PR is merged would need arrow installed.

What do you mean Ubuntu users do not need arrow?  That is the distro that's missing it from the main repos.

Reply:
https://arrow.apache.org/install/ 
contains ubuntu and centos, but not fedora (on my private fedora, I built it from sources)

I meant most users who clone/fork CEPH/RGW but do not need to install arrow-library for their purposes.
the ones that need arrow-library should manually install arrow library.

the RGW code contains the appropriate flags, in case the arrow library is not installed, the arrow-related code is not compiled.

Reply:
If RGW will always need arrow in order to be built, then it doesn't make sense to have the conditional build `if` statements.

The point of `install-deps.sh` is to make sure all build dependencies are installed so `make` will run without extra steps.  I don't agree that users should have to manually enable the repo.

So in install-deps.sh,

1. Get rid of the `if` statements
2. The repo URL is also wrong (that's my bad).  Correct URL is https://apache.jfrog.io/artifactory/arrow/ubuntu
3. The `rm -f` needs `sudo` added and it should also be *after* `if [ "$control" != "debian/control" ] ; then rm $control; fi`

Reply:
+1

Reply:
my only point is
users with fedora machines will fail the arrow installation